iib/workers/tasks/build_add_deprecations.py

```diff
@@ -0,0 +1,198 @@
+        ``binary_image`` to use.
+    """
+    _cleanup()
+    set_request_state(request_id, 'in_progress', 'Resolving the index images')
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigAddDeprecations(
+            _binary_image=binary_image,
+            from_index=from_index,
+            overwrite_from_index_token=overwrite_from_index_token,
+            operator_package=operator_package,
+            deprecation_schema=deprecation_schema,
+            binary_image_config=binary_image_config,
+        ),
+    )
+
+    from_index_resolved = prebuild_info['from_index_resolved']
+    binary_image_resolved = prebuild_info['binary_image_resolved']
+    Opm.set_opm_version(from_index_resolved)
+
+    _update_index_image_build_state(request_id, prebuild_info)
+
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        operators_in_db, index_db_path = verify_operators_exists(
+            from_index_resolved,
+            temp_dir,
+            [operator_package],
+            overwrite_from_index_token,
+        )
+        if not operators_in_db:
+            log.debug(
+                '%s operator package is either not present in index or it is opted in fbc',
+                operator_package,
+            )
+            err_msg = f'Cannot add deprecations for {operator_package}'
+            log.error(err_msg)
+            raise IIBError(err_msg)
+
+        add_deprecations_to_index(
+            request_id,
+            temp_dir,
+            from_index_resolved,
+            operator_package,
+            deprecation_schema,
+            binary_image_resolved,
+            index_db_path,
+        )
+
+        _add_label_to_index(
+            'com.redhat.index.delivery.version',
+            prebuild_info['ocp_version'],
+            temp_dir,
+            'index.Dockerfile',
+        )
+
+        _add_label_to_index(
+            'com.redhat.index.delivery.distribution_scope',
+            prebuild_info['distribution_scope'],
+            temp_dir,
+            'index.Dockerfile',
+        )
+
+        arches = prebuild_info['arches']
+        for arch in sorted(arches):
+            _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
+            _push_image(request_id, arch)
+
+    set_request_state(request_id, 'in_progress', 'Creating the manifest list')
+    output_pull_spec = _create_and_push_manifest_list(request_id, arches, build_tags)
+
+    _update_index_image_pull_spec(
+        output_pull_spec,
+        request_id,
+        arches,
+        from_index,
+        overwrite_from_index,
+        overwrite_from_index_token,
+        from_index_resolved,
+        add_or_rm=True,
+    )
+    _cleanup()
+    set_request_state(
+        request_id, 'complete', 'The deprecation schema was successfully added to the index image'
+    )
+
+
+def add_deprecations_to_index(
```

Comment:
Could you please add new lines and a few comments to make this function more readable?

Reply:
sure I can new lines and comments. Is there anything specific that was hard to go through?